cartoframes/data/dataset/dataset.py

```diff
@@ -1,471 +1,2 @@
-
-            set_default_credentials(
-                base_url='https://your_user_name.carto.com/',
-                api_key='your api key'
-            )
-
-            d = Dataset('tablename')
-            d.update_dataset_info(privacy=Dataset.PRIVACY_PRIVATE)
-    """
-
-    PRIVACY_PUBLIC = DatasetInfo.PRIVACY_PUBLIC
-    """Dataset privacy for datasets that are public.
-
-    Example:
-
-        .. code::
-
-            from cartoframes.data import Dataset, DatasetInfo
-            from cartoframes.auth import set_default_credentials
-
-            set_default_credentials(
-                base_url='https://your_user_name.carto.com/',
-                api_key='your api key'
-            )
-
-            d = Dataset('tablename')
-            d.update_dataset_info(privacy=Dataset.PRIVACY_PUBLIC)
-    """
-
-    PRIVACY_LINK = DatasetInfo.PRIVACY_LINK
-    """Dataset privacy for datasets that are accessible by link.
-
-    Example:
-
-    .. code::
-
-        from cartoframes.data import Dataset, DatasetInfo
-        from cartoframes.auth import set_default_credentials
-
-        set_default_credentials(
-            base_url='https://your_user_name.carto.com/',
-            api_key='your api key'
-        )
-
-        d = Dataset('tablename')
-        d.update_dataset_info(privacy=Dataset.PRIVACY_LINK)
-    """
-
-    def update_dataset_info(self, privacy=None, table_name=None):
-        """Update/change Dataset privacy and name
-
-        Args:
-          privacy (str, optional): One of :py:attr:`Dataset.PRIVACY_PRIVATE`,
-            :py:attr:`Dataset.PRIVACY_PUBLIC` or :py:attr:`Dataset.PRIVACY_LINK`
-          table_name (str, optional): Name of the dataset on CARTO. After updating it,
-            the table_name will be changed too.
-
-        Example:
-
-            .. code::
-
-                from cartoframes.data import Dataset, DatasetInfo
-                from cartoframes.auth import set_default_credentials
-
-                set_default_credentials(
-                    base_url='https://your_user_name.carto.com/',
-                    api_key='your api key'
-                )
-
-                d = Dataset('tablename')
-                d.update_dataset_info(privacy=Dataset.PRIVACY_LINK)
-
-        """
-
-        return self._strategy.update_dataset_info(privacy, table_name)
-
-    def compute_geom_type(self):
-        """Compute the geometry type from the data"""
-        return self._strategy.compute_geom_type()
-
-    def _init_strategy(self, data, credentials=None, schema=None):
-        credentials = credentials or get_default_credentials()
-        for strategy in self._registry.get_strategies():
-            if strategy.can_work_with(data, credentials):
-                return strategy.create(data, credentials, schema)
-
-        raise ValueError('We can not detect the Dataset type')
-
-    def _get_strategies_registry(self):
-        return StrategiesRegistry()
+class Dataset():
```

Comment:
Can we remove this class and rename CatalogDataset to Dataset?

Reply:
True. This can be removed now. I thought about doing the `CatalogDataset` rename in a different PR, but it can be done in this one.